detect_secrets/core/scan.py

```diff
@@ -26,21 +28,58 @@
 from .potential_secret import PotentialSecret
 
 
-def get_files_to_scan(*paths: str, should_scan_all_files: bool) -> Generator[str, None, None]:
-    if not should_scan_all_files:
-        try:
-            valid_paths = git.get_tracked_files(git.get_root_directory())
-        except subprocess.CalledProcessError:
-            log.warning('Did not detect git repository. Try scanning all files instead.')
-            yield from []
-            return
+def get_files_to_scan(
+    *paths: str,
+    should_scan_all_files: bool = False
+) -> Generator[str, None, None]:
+    """
+    If we specify specific files, we should be able to scan them. This abides by the
+    Principle of Least Surprise -- so users don't have to do:
+
+        $ detect-secrets scan test_data/config.env --all-files
+
+    to scan the specific file.
+
+        >>> list(get_files_to_scan('test_data/config.env')) == ['test_data/config.env']
+
+    In a similar way,
+
+        >>> list(get_files_to_scan('test_data/config.env', '.secrets.baseline')) == \
+        ...     ['test_data/config.env', '.secrets.baseline']
+
+    If we specify directories, then we should use git tracked files when possible. To
+    override this behavior, we can specify `should_scan_all_files=True`, which will force
+    the scan for all files.
+
+    See test cases for more details.
+    """
+    # If this is True, then it will consider everything to be valid.
+    # Otherwise, it will only list the files that are valid.
+    valid_paths: Union[bool, Set[str]] = True
+    for path in paths:
+        if not os.path.isdir(path):
```

Comment:
Maybe this would be clearer if you used `if os.path.isdir(path):`

Reply:
added more comments